### PR TITLE
New probe module to perform syn-ack scans

### PR DESCRIPTION
**Pull request by paul-pearce**

### CODE REVIEW

File: src/probe_modules/module_tcp_synackscan.c [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	make_eth_header(eth_header, src, gw);
+	struct ip *ip_header = (struct ip*)(&eth_header[1]);
+	uint16_t len = htons(sizeof(struct ip) + sizeof(struct tcphdr));
+	make_ip_header(ip_header, IPPROTO_TCP, len);
+	struct tcphdr *tcp_header = (struct tcphdr*)(&ip_header[1]);
+	make_tcp_header(tcp_header, dst_port, TH_SYN | TH_ACK);
+	return EXIT_SUCCESS;
+}
+
+void synackscan_process_packet(const u_char *packet,
```
Comment (zakird) [inline src/probe_modules/module_tcp_synackscan.c]: This should probably be static.

File: src/probe_modules/module_tcp_synackscan.c [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+#include "../../lib/includes.h"
+#include "../fieldset.h"
+#include "probe_modules.h"
+#include "packet.h"
+#include "module_tcp_synscan.h"
+
+probe_module_t module_tcp_synackscan;
+
+int synackscan_init_perthread(void* buf, macaddr_t *src,
```
Comment (zakird) [inline src/probe_modules/module_tcp_synackscan.c]: This should probably be static.

File: src/probe_modules/module_tcp_synackscan.c [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+	fs_add_uint64(fs, "dport", (uint64_t) ntohs(tcp->th_dport));
+	fs_add_uint64(fs, "seqnum", (uint64_t) ntohl(tcp->th_seq));
+	fs_add_uint64(fs, "acknum", (uint64_t) ntohl(tcp->th_ack));
+	fs_add_uint64(fs, "window", (uint64_t) ntohs(tcp->th_win));
+	fs_add_uint64(fs, "ipid", (uint64_t) ntohs(ip_hdr->ip_id));
+
+	if (tcp->th_flags & TH_RST) { // RST packet
+		fs_add_string(fs, "classification", (char*) "rst", 0);
+		fs_add_bool(fs, "success", 1);
+	} else { // SYNACK packet
```
Comment (zakird) [inline src/probe_modules/module_tcp_synackscan.c]: Why is a SYN/ACK not a success?

**Comment by zakird** [comment]
Overall this looks good, although there a couple of small nits I would like to see fixed up before merging.

File: src/probe_modules/module_tcp_synackscan.c (hunk 1 above)
Comment (paul-pearce) [inline src/probe_modules/module_tcp_synackscan.c]: Done

File: src/probe_modules/module_tcp_synackscan.c (hunk 2 above)
Comment (paul-pearce) [inline src/probe_modules/module_tcp_synackscan.c]: Done

File: src/probe_modules/module_tcp_synackscan.c (hunk 3 above)
Comment (paul-pearce) [inline src/probe_modules/module_tcp_synackscan.c]: Changed

**Pull request merged**